### DISCUSSION TITLE
ping_interval doesn't change when upgrade to websocket

### DISCUSSION THREAD

**Discussion by RavioliSauce** [miguelgrinberg/python-socketio#896] [Q&A] [answered]
I'll preface this by saying I'm new to websockets. So if I seem misguided, my apologies. I'm changing the ping_interval EngineIO option to 60 minutes, which works fine while long-polling. However if allow_upgrades isn't set to false, every 30 seconds, 64 bytes of data is sent, and 64 is received via NetHogs(which is what I'm using to also monitor when it's long-polling). With this connection being on a limited-data connection, this would be roughly 10.5 megabytes of usage in a month.
I have enabled the logger and EngineIO logger on both the client and the server, yet when data is sent or received, the logger doesn't show anything happening while data is being sent, and received. I've put the output of logger below, not sure that it will help. I could continue to long-poll for the time being, but just found this curious.

You can duplicate this by using the examples here: https://python-socketio.readthedocs.io/en/latest/intro.html

I let this code run for 10 minutes:

SERVER:

import eventlet
import socketio

sio = socketio.Server(ping_interval=3600, logger=True, engineio_logger=True)
app = socketio.WSGIApp(sio)

@sio.event
def connect(sid, environ):
    print('connect ', sid)

@sio.event
def my_message(sid, data):
    print('message ', data)

@sio.event
def disconnect(sid):
    print('disconnect ', sid)

if __name__ == '__main__':
    eventlet.wsgi.server(eventlet.listen(('', 5000)), app)

CLIENT: 

import socketio

sio = socketio.Client(logger=True, engineio_logger=True)

@sio.event
def connect():
    print('connection established')

@sio.event
def my_message(data):
    print('message received with ', data)

@sio.event
def disconnect():
    print('disconnected from server')

sio.connect('http://192.168.1.143:5000')

Client log output(exception at bottom is keyboardinterupt):

Attempting polling connection to http://192.168.1.143:5000/socket.io/?transport=polling&EIO=4
Polling connection accepted with {'sid': 'peZIpz0MtvHvsiQVAAAE', 'upgrades': ['websocket'], 'pingTimeout': 20000, 'pingInterval': 3600000}
Engine.IO connection established
Sending packet MESSAGE data 0{}
Attempting WebSocket upgrade to ws://192.168.1.143:5000/socket.io/?transport=websocket&EIO=4
WebSocket upgrade was successful
Received packet NOOP data 
Received packet MESSAGE data 0{"sid":"oC8B1fHbBNq6vr5PAAAF"}
Namespace / is connected
connection established
Sending packet CLOSE data None
Engine.IO connection dropped
Exiting write loop task
Unexpected error decoding packet: "string index out of range", aborting
Waiting for write loop task to end
Exiting read loop task
Exception ignored in: <module 'threading' from '/usr/lib/python3.7/threading.py'>
Traceback (most recent call last):
  File "/usr/lib/python3.7/threading.py", line 1281, in _shutdown
    t.join()
  File "/usr/lib/python3.7/threading.py", line 1032, in join
    self._wait_for_tstate_lock()
  File "/usr/lib/python3.7/threading.py", line 1048, in _wait_for_tstate_lock
    elif lock.acquire(block, timeout):
  File "/home/orangepi/.local/lib/python3.7/site-packages/engineio/client.py", line 36, in signal_handler
    return original_signal_handler(sig, frame)
  File "/home/orangepi/.local/lib/python3.7/site-packages/socketio/client.py", line 26, in signal_handler
    return original_signal_handler(sig, frame)
KeyboardInterrupt
disconnected from server

**Comment by miguelgrinberg** [comment, marked as answer]
Please follow the instructions. If you don't know if what you are reporting is an issue, then don't write an issue, open a discussion topic. The issues board is for tracking issues, not to troubleshoot.

As you can see in the log that you provided, no ping/ponging happened during your test. My guess is that this is caused by keepalive packets sent by the Websocket server that you are using (eventlet), and completely unrelated and outside of the control of this package. Have you looked at the data being exchanged? If don't think you'll find Socket.IO ping/pong packets in there, since clearly these aren't being exchanged, according to the log.

**Reply by RavioliSauce** [reply]
My apologies again, and thank you for responding.
I have found in eventlet a parameter "keepalive=True". However making false still sends these packets every 30 seconds. I'm not sure how to see the data being exchanged, so I can't tell exactly what function from eventlet is causing this. I suppose for now, I'll just stick with long-polling as that's the only way I can truly control how many times the server pings. Thanks for your time, I greatly appreciate it
Edit: Also I will be sure to open a discussions topic next time

**Reply by miguelgrinberg** [reply]
The problem is that these mechanisms that you are trying to disable are what makes the protocol reliable. When you set the `ping_interval` to one hour, it means that clients that are abruptly disconnected will not be detected for up to one hour, when the next ping happens. I understand your reasons, but this is going to make your solution extremely unreliable.

**Reply by RavioliSauce** [reply]
Understood. Do you have any suggestions or recommendations on how to reduce idle data usage? Other than changing the ping_interval? Maybe reducing ping size if thats even a thing

**Reply by miguelgrinberg** [reply]
Well, Socket.IO is a high-level protocol. If you need to have so much control over the traffic, then you should probably manage the WebSocket connection yourself.
